tests/integration/plugins/test_python.py

```diff
@@ -408,3 +409,54 @@ def _get_package_install_commands(self) -> list[str]:
 
     primed_script = lf.project_info.prime_dir / "bin/mytest"
     assert not primed_script.exists()
+
+
+@pytest.mark.slow
+def test_python_plugin_get_files(new_dir, partitions):
+    parts_yaml = textwrap.dedent(
+        """\
+        parts:
+          foo:
+            plugin: python
+            source: .
+            python-packages: [flask==3.1.0]
+        """
+    )
+    parts = yaml.safe_load(parts_yaml)
+
+    lifecycle = LifecycleManager(
+        parts,
+        application_name="test_python",
+        cache_dir=new_dir,
+        partitions=partitions,
+    )
+    actions = lifecycle.plan(Step.BUILD)
+
+    with lifecycle.action_executor() as ctx:
+        ctx.execute(actions)
+
+    part_name = list(parts["parts"].keys())[0]
+    actual_file_list = lifecycle._executor._handler[part_name]._plugin.get_files()
```

Comment:
Could you instantiate a separate plugin here and run `get_files()` on it to check that it's stateless? (I know it is now - this just ensures that future changes don't accidentally make it stateful)

Reply:
https://github.com/canonical/craft-parts/pull/943/commits/1528104367e622ef56199cc2bc1a406983bce0e6